### PR TITLE
Add KafkaStreamsConfiguration

### PR DESCRIPTION
**Pull request by garyrussell**
See https://github.com/spring-projects/spring-boot/pull/14021#discussion_r208929624

Avoid using a bean of type `Properties`.

### CODE REVIEW

**Comment by artembilan** [comment]
Well, with this I believe we need to change Spring Integration global properties from the `Properties` bean as well...

File: spring-kafka/src/main/java/org/springframework/kafka/config/KafkaStreamsConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.kafka.streams.StreamsBuilder;
+
+import org.springframework.util.Assert;
+
+/**
+ * Wrapper for {@link StreamsBuilder} properties; used to avoid auto-configuring
+ * a generic {@link Properties} object in Boot.
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/config/KafkaStreamsConfiguration.java]: I don't think this is a good idea to mentioned Spring Boot here.
More over in its non-official name.

I believe we can find some generic message to say here.

Thanks

**Comment by garyrussell** [comment]
>Well, with this I believe we need to change Spring Integration global properties from the Properties bean as well...

Yes, then I guess we can expose the props as boot auto config properties.

File: spring-kafka/src/main/java/org/springframework/kafka/config/KafkaStreamsConfiguration.java (hunk 1 above)
Comment (garyrussell) [reply spring-kafka/src/main/java/org/springframework/kafka/config/KafkaStreamsConfiguration.java]: Good catch on the missed ctor.

**Pull request merged**